benchexec/tools/hornix.py

```diff
@@ -21,7 +23,10 @@ def executable(self, tool_locator):
         return tool_locator.find_executable("hornix")
 
     def version(self, executable):
-        return self._version_from_tool(executable, arg="--version")
+        output = self._version_from_tool(executable, arg="--version")
+        version_pattern = r"\d+\.\d+\.\d+"
```

Comment:
Is this complete and future-proof? I.e., could it happen that your tool eventually produces version numbers with not exactly 3 digits? Something like `10.0.0` or `3.1.4.2` or `3.3.3-beta1`?

I would recommend to make this more general.

Reply:
First example should be covered by this regex already (note the `+`).
From the second and third this would extract `3.1.4` and `3.3.3`, respectively. That is not ideal.
But we plan to stick to the main semantic versioning.

It would help us to get this merged as soon as possible, so we could actually get a pre-run today.

Reply:
Please provide a follow-up PR until tomorrow such that it can get into the next release of BenchExec.